docs/reference/contrib/redirects.rst

```diff
@@ -40,6 +40,25 @@ Page model recipe of to have redirects created automatically when changing a pag
 
 For an editor's guide to the interface, see :ref:`managing_redirects`.
 
+Automatic redirect creation
+===========================
+
+.. versionadded:: 2.16
+
+Wagtail comes with an option to automatically create permanent redirects for pages (and their descendants) when they are moved or have their slug changed. This feature is disabled by default, but you can enable it for your project by adding the following to your settings:
+
+.. code-block:: python
+
+  WAGTAILREDIRECTS_AUTOCREATE = True
```

Comment:
I think this feature should be enabled by default. Otherwise, I feel many people won't enable this because they wouldn't know it existed. Very few sites out there override the URL generation (I've never come across one personally), and any that do have problems will likely look for this setting and switch it off.

I'm also keen to not require new users to have to read about all of Wagtail's settings in order to get the best experience. I think we should enable the best experience by default, and allow people to switch things off it they want to.

Reply:
Those are all good points, but I am a little concerned about the time lag it might add for larger sites. I would feel far more comfortable enabling it by default if we had the background workers stuff in place, and there wouldn't be a noticeable impact to editors.

Reply:
Plus, the RFC specifies it should be opt-in.

Reply:
In client sites, I have this running in a separate thread so that it's not as much of an issue - but I thought that would be taking things in a strange direction for Wagtail with the background worker stuff on the horizon.

Reply:
> Those are all good points, but I am a little concerned about the time lag it might add for larger sites. I would feel far more comfortable enabling it by default if we had the background workers stuff in place - then, we could do this without a noticeable impact to editors.

I feel like moving thousands of pages is a very rare operation and editors doing this probably won't find a few second lag to be unexpected. So I'm not sure if disabling it everywhere to make this very uncommon operation faster is a good tradeoff?

Do we have any benchmarks on how long it takes to bulk create redirects? (I'd expect  it to be 1000s/second as the model is quite simple)

Reply:
My standpoint is a protective one: Wagtail is not the most performant of CMSs (fact), and I don't want to give people reasons to hate on it even more. Plus - what about all the headless CMSs out there to which redirects are probably irrelevant? At least if a developer enables it, they can explain the speed trade-off.

Reply:
No official benchmarks, no. But, I did hit a few crashes on Tate - mostly due to the URL method overrides on that site to mimic those used in their digital archives.

Reply:
> and editors doing this probably won't find a few second lag to be unexpected.

I have to disagree with you here. Making users wait for any longer than a second without telling them why is bad practice. They're aiming to include a prompt in the new UI to warn users when making a change that will impact the URL - in which case is may be acceptable. But, we're just not there yet.

I'm definitely in favour of making it the default behaviour at a later time - but maybe let's just see how it goes? No code is forever :)